src/client.js

```diff
@@ -11,9 +11,29 @@ import hashLinkScroll from './utils/hash-link-scroll'
 import loggerFactory from './logger'
 import releaseBranchConsts from '@twreporter/core/lib/constants/release-branch'
 import twreporterRedux from '@twreporter/redux'
+// lodash
+import get from 'lodash/get'
+const _ = {
+  get,
+}
 
 const logger = loggerFactory.getLogger()
 const releaseBranch = globalEnv.releaseBranch
+const tagManagerArgs = {
+  master: {
+    gtmId: 'GTM-PRMXBBN',
+    auth: '2pJC7GotZqWa7HtmIgSFIg',
```

Comment:
Could you share the idea about why do we need `auth`, `preview` and why do they are different corresponds to different environments (master, staging) ?

Reply:
The official doc about environments can be found [here](https://support.google.com/tagmanager/answer/6311518).

The main reason we need to include environments is to publish testing gtm version to testing environment, thus we can show test data on GA matrix.
I separate master & staging for corresponding to releaseBranch in codebase. They can be combined into a single testing environment.

How do you consider about it ?

Reply:
I second this idea.

We can separate master & staging as different environments.